### PR TITLE
[2.5.8] Add global seLinux toggle to rancher-logging

### PR DESCRIPTION
**Pull request by cmurphy**
Add a new value to the rancher-logging chart to enable setting the
seLinuxOptions securityContext attribute for Logging resources and the
rke-aggregator DaemonSet. This is applied to any resource that needs to
read from /var/log/containers or /var/lib/docker/containers.

Without this patch, components in the rancher-logging chart mount and
attempt to read and sometimes write files on several hostPaths, and when
SELinux is enabled on the host, will fail because the containers are
unauthorized to do so. The added seLinuxOptions attribute in this patch
causes the fluentbit and rke-aggregator DaemonSets to run with the
SELinux context type `rke_logreader_t`, which is provided by the
rke2-linux package and enables the container to read files with the
label `container_log_t`, which includes files like
/var/log/containers/*.log, and files with type `rke_container_var_lib_t`
which the docker log files are relabeled to.

The logging aggregator will still fail to read and write log data from
/var/lib/rancher/logging, but this will be addressed in a different way
(https://github.com/rancher/rancher/issues/31309).

https://github.com/rancher/rancher/issues/30949

Depends on https://github.com/rancher/rke2-selinux/pull/13

### CODE REVIEW

**Review by nickgerace — DISMISSED** [review]
LGTM, awesome

**Review by dweomer — APPROVED** [review]

File: packages/rancher-logging/package.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 url: https://kubernetes-charts.banzaicloud.com/charts/logging-operator-3.9.0.tgz
 packageVersion: 00
-releaseCandidateVersion: 08
+releaseCandidateVersion: 09
```
Comment (cbron) [inline packages/rancher-logging/package.yaml:3]: Once rest of PR is confirmed you'll need to pull in upstream again and bump packageVersion.

File: packages/rancher-logging/generated-changes/patch/README.md.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--- charts-original/README.md
++++ charts/README.md
+@@ -67,6 +67,7 @@
+ | `securityContext`                                   | Container SecurityContext for Logging operator. [More info](https://kubernetes.io/docs/concepts/policy/security-context/) | `{"allowPrivilegeEscalation": false, "readOnlyRootFilesystem": true}` |
+ | `createCustomResource`                              | Create CRDs. | `true` |
+ | `monitoring.serviceMonitor.enabled`                 | Create Prometheus Operator servicemonitor. | `false` |
++| `global.seLinux.enabled`                            | Add seLinuxOptions to Logging resources | `false` |
```
Comment (cbron) [inline packages/rancher-logging/generated-changes/patch/README.md.patch:7]: We should add that it requires the RPM. Could also link out to docs once they are ready.

**Review by paynejacob — COMMENTED** [review]
We should add a note about the rpm dependency and then I think it is good.

File: packages/rancher-logging/generated-changes/patch/README.md.patch (hunk 2 above)
Comment (cmurphy) [reply packages/rancher-logging/generated-changes/patch/README.md.patch]: Added a link to the rke2-selinux github release page

File: packages/rancher-logging/package.yaml (hunk 1 above)
Comment (cmurphy) [reply packages/rancher-logging/package.yaml]: Bumped the packageVersion, does the releaseCandidateVersion stay the same or get reset?

Comment (cbron) [reply packages/rancher-logging/package.yaml]: Needs to get reset to 01

File: packages/rancher-logging/generated-changes/patch/README.md.patch (hunk 2 above)
Comment (cbron) [reply packages/rancher-logging/generated-changes/patch/README.md.patch]: That works for now, we should try to remember to swap once it moves, ideally to the official docs.

File: packages/rancher-logging/package.yaml (hunk 1 above)
Comment (cmurphy) [reply packages/rancher-logging/package.yaml]: done

**Review by cbron — APPROVED** [review]

**Pull request merged**